barretenberg/cpp/src/barretenberg/bb/main.cpp

```diff
@@ -537,34 +537,33 @@ void avm_prove(const std::filesystem::path& bytecode_path,
     // TODO(ilyas): <#4887>: Currently we only need these two parts of the vk, look into pcs_verification key reqs
     std::vector<uint64_t> vk_vector = { verification_key.circuit_size, verification_key.num_public_inputs };
 
-    std::filesystem::path output_vk_path = output_path.parent_path() / "vk";
-    write_file(output_vk_path, to_buffer(vk_vector));
-    write_file(output_path, to_buffer(proof));
+    write_file(output_path / "proof", to_buffer(proof));
+    write_file(output_path / "vk", to_buffer(vk_vector));
 }
 
 /**
  * @brief Verifies an avm proof and writes the result to stdout
  *
  * Communication:
- * - stdout: The boolean value indicating whether the proof is valid.
  * - proc_exit: A boolean value is returned indicating whether the proof is valid.
+ *   an exit code of 0 will be returned for success and 1 for failure.
  *
- * @param proof_path Path to the file containing the serialised proof (the vk should also be in the parent)
+ * @param proof_path Path to the file containing the serialized proof
+ * @param vk_path Path to the file containing the serialized verification key
+ * @return true If the proof is valid
+ * @return false If the proof is invalid
  */
-bool avm_verify(const std::filesystem::path& proof_path)
+bool avm_verify(const std::filesystem::path& proof_path, const std::filesystem::path& vk_path)
 {
-    std::filesystem::path vk_path = proof_path.parent_path() / "vk";
     std::vector<fr> const proof = many_from_buffer<fr>(read_file(proof_path));
     std::vector<uint8_t> vk_bytes = read_file(vk_path);
     auto circuit_size = from_buffer<size_t>(vk_bytes, 0);
     auto num_public_inputs = from_buffer<size_t>(vk_bytes, sizeof(size_t));
     auto vk = AvmFlavor::VerificationKey(circuit_size, num_public_inputs);
 
-    std::cout << avm_trace::Execution::verify(vk, proof);
-    return avm_trace::Execution::verify(vk, proof);
-
-    std::cout << 1;
-    return true;
+    const bool verified = avm_trace::Execution::verify(vk, proof);
+    vinfo("verified: ", verified);
+    return verified;
 }
```

Comment:
We were verifying twice?! :eyes:

Reply:
We wanted to be VERY sure the proof was ok 😆